mapper.go

```diff
@@ -107,40 +118,105 @@ func (m *metricMapper) initFromString(fileContents string) error {
 
 	m.mutex.Lock()
 	defer m.mutex.Unlock()
-	m.mappings = parsedMappings
+	m.Mappings = parsedMappings
 
 	mappingsCount.Set(float64(len(parsedMappings)))
 
 	return nil
 }
 
+func (m *metricMapper) initFromYAMLString(fileContents string) error {
+	var n metricMapper
+
+	if err := yaml.Unmarshal([]byte(fileContents), &n); err != nil {
+		return err
+	}
+
+	if n.Defaults.Buckets == nil || len(n.Defaults.Buckets) == 0 {
+		n.Defaults.Buckets = prometheus.DefBuckets
+	}
+
+	for i := range n.Mappings {
+		currentMapping := &n.Mappings[i]
+
+		if !metricLineRE.MatchString(currentMapping.Match) {
+			return fmt.Errorf("invalid match: %s", currentMapping.Match)
+		}
+
+		// check that label is correct
+		for k, v := range currentMapping.Labels {
+			label := fmt.Sprintf("%s=%q", k, v)
+			if len(labelLineRE.FindStringSubmatch(label)) != 3 {
```

Comment:
It is a bit complicated to first make a k=v string out of this and then match the entire string against `labelLineRE`. The label value can be anything, so it would be better to just check whether the label name (`k`) matches `identifierRE`.